arm64 prl-tools not compiling on kernel v6.5.x

### Describe the bug
Attempting to upgrade to kernel v6.5.x results in the arm64 prl-tools package (18.3.0-53606) failing to compile.

### Steps To Reproduce
Steps to reproduce the behavior:
1. Upgrade NixOS in a Parallels VM from kernel version v6.4.x to v6.5.x

### Expected behavior
nixos-rebuild to complete without error

### Screenshots
```console
unpacking channels...
building Nix...
building the system configuration...
these 39 derivations will be built:
  /nix/store/srkp0j9pz314xgdrf0m9ki9cc83jsafb-ParallelsDesktop-19.0.0-54570.dmg.drv
  /nix/store/v19vj4g18a3wmk4c66bgfadafyqiw40k-uname.drv
  /nix/store/f8k707jqjfp45qmlrg7lgy7vra7xfbjv-prl-tools-19.0.0-54570.drv
  /nix/store/i132rjx3wyf2swdv79xqw1qkkhgliz2k-prl-tools-18.3.0-53606.drv
  /nix/store/6vx9vy5j807rd6qhbr3rrmb722kpjnbv-system-path.drv
  /nix/store/mh0a0w8b3ncq0s153bdxichkij7rd3mn-set-environment.drv
  /nix/store/2a3wvy0y1hfbzr4l2cxk3zrwz3qi140y-etc-zshenv.drv
  /nix/store/dkxz4bddmgac49lc4y8a4f28v0dmwcss-dbus-1.drv
  /nix/store/sv9yw632x5ac191w92vkbbwa1xcyfxs5-X-Restart-Triggers.drv
  /nix/store/2mvmzqizdh4czb7qi71ln1kfn3xj7pw7-unit-dbus.service.drv
  /nix/store/52zsa6ilyjq80k1ylyrlx3kn83s20nqg-unit-prlcp.service.drv
  /nix/store/6gxgcsp08wnv2xcqwflwfcm7aaxkryq9-unit-dbus.service.drv
  /nix/store/70hdbjx00jp4pbx24hf9v4dv1yqa03x4-etc-environment.drv
  /nix/store/82rca044dz0xq3zki341rwnaa82ahf5l-udev-rules.drv
  /nix/store/v76vynr86x5qaj3xs6s0v9rjwqm908lw-linux-6.5.8-modules.drv
  /nix/store/kpq798qw1n0ifcbbmh64d5byrc1jw4dq-linux-6.5.8-modules-shrunk.drv
  /nix/store/912a004m4l8wm96v960gn8mz7yln9s2b-stage-1-init.sh.drv
  /nix/store/a0a0nl2bx08r6v95wddm2kga2fyypsdk-unit-prlfsmountd.service.drv
  /nix/store/aljv3xfh5kzhnp9jhczz124d6wvkvyhn-unit-prltoolsd.service.drv
  /nix/store/aw19c5dgckmjdq80rsyxlj5159fzxdbn-unit-prlshprint.service.drv
  /nix/store/i9nj182hq4pbkypydn0i2nkfjhjvk4zk-X-Restart-Triggers.drv
  /nix/store/d828l3k9dqnvh6p0xbxmq7crarass9z6-unit-systemd-udevd.service.drv
  /nix/store/f9xr8sb8dg39xnygcnnsrbax3iwhfzw3-X-Restart-Triggers.drv
  /nix/store/gn5kqsmg8fl5dw33rk210xvdwrfn8nmi-etc-profile.drv
  /nix/store/hp45szk0b7vnl3p4x6iimxlf263k03hv-unit-systemd-fsck-.service.drv
  /nix/store/lm7ywrn2pqd5h14r278rjmgjj6zg1rpx-unit-polkit.service.drv
  /nix/store/vpi9yfxh08l7r23mhn2bjw62zk9rvw7d-unit-accounts-daemon.service.drv
  /nix/store/qnm5s7lqh3i0dhvvhwj4wgc2yx7xpjdn-system-units.drv
  /nix/store/qphwixxgnrw2npnfni0x59h8lliximgk-hwdb.bin.drv
  /nix/store/i7hx6i99ldh0w6z0zqdl9hfh0f31m5ly-unit-prlshprof.service.drv
  /nix/store/lv2amcylcs2mc69s6sn732vk5z5rwjvy-unit-prldnd.service.drv
  /nix/store/mdabil0l65vzh23jrm5ns5w8dlnwxss3-unit-prlsga.service.drv
  /nix/store/v0y139nxg075ja6h0drpz6s65f4wlxbi-unit-prlcc.service.drv
  /nix/store/yxdr0p3p3anv0dghdid4r61iyw3n403y-user-units.drv
  /nix/store/j9pzy43ilrhndnlcxw1wlyggw1qpqpk1-etc.drv
  /nix/store/sk6mxbgzy8xq60j23qlr11ld8kvvb5k0-initrd-linux-6.5.8.drv
  /nix/store/wjf30nda664acq4sljqvqsc24s24lhlx-boot.json.drv
  /nix/store/xkb8g249hyl6hnwia03nxija1n1wvyxa-kernel-modules.drv
  /nix/store/fkbdkvnpgkgzfk42qrj1npdkj7qcwlbk-nixos-system-parallels-work-mac-mini-23.05.4448.5550a85a087c.drv
these 14 paths will be fetched (136.40 MiB download, 604.16 MiB unpacked):
  /nix/store/042vhwb2vhpmnb1xxm5kqw7zkddk5z9m-auto-patchelf-hook
  /nix/store/2jxvz8rarsil879x4jqxkh0m81hmndj9-python3.10-pyelftools-0.29
  /nix/store/48yl626razzri9vaqpz2lqf76xmzra4q-zstd-1.5.5-dev
  /nix/store/5g1nad05pcvaqiy1z4q9s07wgq13nyrs-util-linux-2.39.2-swap
  /nix/store/693qkqll3ajawqpsxqsrarccfgd8a7dj-linux-6.5.7-dev
  /nix/store/c2w57rryjzdr9lj4b481r0zcm05yczqx-timetrap-1.15.2
  /nix/store/i20hx6f3lfgxvd3hlw7fdv36k8dr9mzy-dbus-glib-0.112-dev
  /nix/store/ibsrvkljjzhiiiixsll22ck36i0dnsy0-glib-2.76.4-dev
  /nix/store/k66z8z7pxli2x7nn0mcsagfgkc5g53xc-python3-3.10.12-env
  /nix/store/lz53c0cjfmjp2ihp50d350cm4xmr4r2a-util-linux-2.39.2-bin
  /nix/store/n9kvfq52fw88f6pf2prgdz9zb2c5aij4-timetrap-1.15.2
  /nix/store/pq2raggppq3f0jzi8cb4mpv5hcgya616-ruby3.1.4-timetrap-1.15.2
  /nix/store/zsba41b85yzfqqa06mxl3p9k6vy1hbq8-util-linux-2.39.2-mount
  /nix/store/zw3z78yfq7gb68hx872r12s94f7g3rcq-timetrap
building '/nix/store/i132rjx3wyf2swdv79xqw1qkkhgliz2k-prl-tools-18.3.0-53606.drv'...
copying path '/nix/store/ibsrvkljjzhiiiixsll22ck36i0dnsy0-glib-2.76.4-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/693qkqll3ajawqpsxqsrarccfgd8a7dj-linux-6.5.7-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/48yl626razzri9vaqpz2lqf76xmzra4q-zstd-1.5.5-dev' from 'https://cache.nixos.org'...
building '/nix/store/v19vj4g18a3wmk4c66bgfadafyqiw40k-uname.drv'...
copying path '/nix/store/5g1nad05pcvaqiy1z4q9s07wgq13nyrs-util-linux-2.39.2-swap' from 'https://cache.nixos.org'...
copying path '/nix/store/zsba41b85yzfqqa06mxl3p9k6vy1hbq8-util-linux-2.39.2-mount' from 'https://cache.nixos.org'...
copying path '/nix/store/2jxvz8rarsil879x4jqxkh0m81hmndj9-python3.10-pyelftools-0.29' from 'https://cache.nixos.org'...
copying path '/nix/store/pq2raggppq3f0jzi8cb4mpv5hcgya616-ruby3.1.4-timetrap-1.15.2' from 'https://cache.nixos.org'...
building '/nix/store/srkp0j9pz314xgdrf0m9ki9cc83jsafb-ParallelsDesktop-19.0.0-54570.dmg.drv'...
unpacking sources
copying path '/nix/store/k66z8z7pxli2x7nn0mcsagfgkc5g53xc-python3-3.10.12-env' from 'https://cache.nixos.org'...
copying path '/nix/store/lz53c0cjfmjp2ihp50d350cm4xmr4r2a-util-linux-2.39.2-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/042vhwb2vhpmnb1xxm5kqw7zkddk5z9m-auto-patchelf-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/n9kvfq52fw88f6pf2prgdz9zb2c5aij4-timetrap-1.15.2' from 'https://cache.nixos.org'...
copying path '/nix/store/i20hx6f3lfgxvd3hlw7fdv36k8dr9mzy-dbus-glib-0.112-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/c2w57rryjzdr9lj4b481r0zcm05yczqx-timetrap-1.15.2' from 'https://cache.nixos.org'...
copying path '/nix/store/zw3z78yfq7gb68hx872r12s94f7g3rcq-timetrap' from 'https://cache.nixos.org'...

trying https://download.parallels.com/desktop/v19/19.0.0-54570/ParallelsDesktop-19.0.0-54570.dmg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  292M  100  292M    0     0  64.6M      0  0:00:04  0:00:04 --:--:-- 66.3M

7-Zip [64] 17.05 : Copyright (c) 1999-2021 Igor Pavlov : 2017-08-28
p7zip Version 17.05 (locale=C,Utf16=off,HugeFiles=on,64 bits,8 CPUs LE)

Scanning the drive for archives:
1 file, 32466944 bytes (31 MiB)

Extracting archive: Parallels Desktop.app/Contents/Resources/Tools/prl-tools-lin-arm.iso
--
Path = Parallels Desktop.app/Contents/Resources/Tools/prl-tools-lin-arm.iso
Type = Iso
Physical Size = 32466944
Created = 2023-05-10 16:23:43
Modified = 2023-05-10 16:23:43

Everything is Ok

Folders: 43
Files: 111
Size:       31791221
Compressed: 32466944
setting SOURCE_DATE_EPOCH to timestamp 1683735815 of file prl-tools-build/tools/tools-arm64/bin/prltimesync
patching sources
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
building
Will use compiler CC="gcc".
Compiler version "gcc (GCC) 12.2.0"
cd prl_tg/Toolgate/Guest/Linux/prl_tg && make CC=gcc
make[1]: Entering directory '/build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg'
1: /nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/build
2: /build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg
3: 20231025
4: 1.10.2
Start compile prl_tg...
make -C /nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/build M=/build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg SRCROOT=/build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg CC=gcc modules
make[2]: Entering directory '/nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/build'
1: /nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/build
2: /build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg
3: 20231025
4: 1.10.2
  CC [M]  /build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg/prltg.o
  CC [M]  /build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg/prltg_call.o
In file included from /build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg/prltg_common.h:25,
                 from /build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg/prltg_call.c:11:
/build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg/prltg_call.c: In function 'tg_req_map_user_pages':
/build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg/prltg_compat.h:63:17: error: too many arguments to function 'get_user_pages'
   63 |                 get_user_pages(_1, _2, (_3) ? FOLL_WRITE : 0, _4, _5)
      |                 ^~~~~~~~~~~~~~
/build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg/prltg_call.c:126:15: note: in expansion of macro 'prl_get_user_pages'
  126 |         got = prl_get_user_pages(
      |               ^~~~~~~~~~~~~~~~~~
In file included from /nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/source/include/linux/scatterlist.h:8,
                 from /nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/source/include/linux/dmapool.h:14,
                 from /nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/source/include/linux/pci.h:1621,
                 from /build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg/prltg_call.c:7:
/nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/source/include/linux/mm.h:2430:6: note: declared here
 2430 | long get_user_pages(unsigned long start, unsigned long nr_pages,
      |      ^~~~~~~~~~~~~~
make[4]: *** [/nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/source/scripts/Makefile.build:243: /build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg/prltg_call.o] Error 1
make[3]: *** [/nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/source/Makefile:2034: /build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg] Error 2
make[2]: *** [/nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/source/Makefile:234: __sub-make] Error 2
make[2]: Leaving directory '/nix/store/mwbl8czdxmijp34igyq4v7v9b3d7wibf-linux-6.5.8-dev/lib/modules/6.5.8/build'
make[1]: *** [Makefile:44: prl_tg] Error 2
make[1]: Leaving directory '/build/prl-tools-build/kmods/prl_tg/Toolgate/Guest/Linux/prl_tg'
make: *** [Makefile.kmods:49: compile] Error 2
error: builder for '/nix/store/i132rjx3wyf2swdv79xqw1qkkhgliz2k-prl-tools-18.3.0-53606.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/i9nj182hq4pbkypydn0i2nkfjhjvk4zk-X-Restart-Triggers.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qphwixxgnrw2npnfni0x59h8lliximgk-hwdb.bin.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xkb8g249hyl6hnwia03nxija1n1wvyxa-kernel-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/v76vynr86x5qaj3xs6s0v9rjwqm908lw-linux-6.5.8-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6vx9vy5j807rd6qhbr3rrmb722kpjnbv-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/v0y139nxg075ja6h0drpz6s65f4wlxbi-unit-prlcc.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/52zsa6ilyjq80k1ylyrlx3kn83s20nqg-unit-prlcp.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lv2amcylcs2mc69s6sn732vk5z5rwjvy-unit-prldnd.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/a0a0nl2bx08r6v95wddm2kga2fyypsdk-unit-prlfsmountd.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/mdabil0l65vzh23jrm5ns5w8dlnwxss3-unit-prlsga.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/aw19c5dgckmjdq80rsyxlj5159fzxdbn-unit-prlshprint.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/i7hx6i99ldh0w6z0zqdl9hfh0f31m5ly-unit-prlshprof.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/aljv3xfh5kzhnp9jhczz124d6wvkvyhn-unit-prltoolsd.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fkbdkvnpgkgzfk42qrj1npdkj7qcwlbk-nixos-system-parallels-work-mac-mini-23.05.4448.5550a85a087c.drv' failed to build
```

### Additional context
- Using `hardware.parallels.enable = false;` allows an upgrade to succeed with no tools installed.
- Attempting an upgrade with the unstable version of `pkgs.linuxKernel.packages.linux_6_5.prl-tools` (19.0.0-54570) results in a similar build error.

### Notify maintainers
@wegank @catap 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-linux"`
 - host os: `Linux 6.5.8, NixOS, 23.05 (Stoat), 23.05.4448.5550a85a087c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.6`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/os-specific/linux/prl-tools/default.nix
+++ b/pkgs/os-specific/linux/prl-tools/default.nix
@@ -36,13 +36,13 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "prl-tools";
-  version = "19.0.0-54570";
+  version = "19.1.0-54729";
 
   # We download the full distribution to extract prl-tools-lin.iso from
   # => ${dmg}/Parallels\ Desktop.app/Contents/Resources/Tools/prl-tools-lin.iso
   src = fetchurl {
     url = "https://download.parallels.com/desktop/v${lib.versions.major finalAttrs.version}/${finalAttrs.version}/ParallelsDesktop-${finalAttrs.version}.dmg";
-    hash = "sha256-y7UC+E5i2cxkOJ9nVI6aQAFJ5kTXv9uaZoMO4/SCS6k=";
+    hash = "sha256-Qo/6EuILu5i3hjhxecf3qfoMFEJHJZVY19tDIXDJiag=";
   };
 
   hardeningDisable = [ "pic" "format" ];

